### ISSUE TITLE
Use io.Reader

### ISSUE THREAD

**Issue by rwcarlsen** [wtolson/go-taglib#1]
Using the Go stdlib io.Reader interface instead of just file opening would make this package more flexible and more idiomatic.

**Comment by wtolson** [comment]
I agree, but not sure how you would go about it as the underlying taglib api only reads from disk. It could write the io.Reader to a temp file to give to taglib but would have great performance characteristics. Any thoughts on how to do this in an efficient manner?

**Comment by rwcarlsen** [comment]
Hm - probably not worth the trouble then.  You could rewrite taglib in go :P

**Comment by GwynethLlewelyn** [comment]
> Hm - probably not worth the trouble then. You could rewrite taglib in go :P

Now **that** would be _the_ thing!! 🤣
